source/dub/dub.d

```diff
@@ -603,15 +599,33 @@ class Dub {
 						import std.stdio;
 						import core.runtime;
 
-						void main() { writeln("All unit tests have been run successfully."); }
-						shared static this() {
+						// libraries need a main function
+						version (dub_test_main) void main() {}
+
+						bool defaultUnitTester() {
+							import core.stdc.stdlib : exit;
+							foreach (modul; allModules)
+								foreach (test; __traits(getUnitTests, modul))
+									test();
+							writeln("All unit tests have been run successfully.");
+							exit(0);
+							return true;
```

Comment:
Should be `return false;`, so that the application main function isn't executed afterwards. That makes it possible to remove the `exit(0);` and solves the module destructor issue.